src/observer/ingest_logic.py

```diff
@@ -794,9 +799,17 @@ def download_regenerate_article(msid):
         log = LOG.debug if err.response.status_code == 404 else LOG.error
         log("failed to fetch article %s: %s", msid, err) # probably an unpublished article.
 
+    except KeyboardInterrupt as exc:
+        raise exc
+
     except BaseException:
         LOG.exception("unhandled exception attempting to download and regenerate article %s", msid)
 
+def download_regenerate_article_list(some_fn):
+    with patch('observer.ingest_logic.content_descriptions', {models.LAX_AJSON: {'api-list': 'articles'}}):
```

Comment:
This is totally ghetto. pass `content_descriptions` as an optional parameter to `download_all`. That is still kinda crap, but less than this and is much more testable.

Reply:
docstr